### PR TITLE
chore: update eslint config

### PR DESCRIPTION
**Pull request by ryanwolhuter**
Update lint script to include tsx files
Update configs to use latest rules
Apply fixes caught by linter

Relevant issue [here](https://github.com/orgs/powerhouse-inc/projects/1/views/1?pane=issue&itemId=48267159)

### CODE REVIEW

File: editors/document-model/editor-schema.tsx [hunk 1]
```diff
@@ -43,12 +43,15 @@ const normalizeSchema = (
...
         // Update type name value if name changed
-        const newCodeValue = schema.replace(typeRegexp, (match, group1) => {
-            return match.replace(
-                group1,
-                `${pascalCase(documentName)}${scopeStateName}State`,
-            );
-        });
+        const newCodeValue = schema.replace(
+            typeRegexp,
+            (match, group1: string) => {
```
Comment (ryanwolhuter) [inline editors/document-model/editor-schema.tsx:48]: the type of `group1` was not being inferred so I annotated it as `string`

File: editors/document-model/editor-schema.tsx [hunk 2]
```diff
@@ -136,8 +139,11 @@ export default function EditorSchema({
         codegen(code).then(result => {
             import(/* @vite-ignore */ result).then(validators => {
                 const schemaName = `${name}StateSchema`;
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
```
Comment (ryanwolhuter) [inline editors/document-model/editor-schema.tsx:142]: it seems like the types of this code are out of our control, so I suppressed these rules here and let them stay as `any`

File: editors/document-model/editor-schema.tsx [hunk 3]
```diff
@@ -161,7 +167,7 @@ export default function EditorSchema({
     return (
         <div>
             <Editor
-                theme={`vs-${props}`}
+                theme={`vs-${theme}`}
```
Comment (ryanwolhuter) [inline editors/document-model/editor-schema.tsx:170]: This was a bug that the linter caught

File: editors/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "../tsconfig.json",
+    "compilerOptions": {
```
Comment (ryanwolhuter) [inline editors/tsconfig.json:3]: extends the main tsconfig to include dom libs in the editors. dom libs are not included in the document models. I checked that the behavior is indeed different in each directory

**Review by gpuente — APPROVED** [review]
LGTM 🚀

**Pull request merged**